security/src/main/java/no/difi/move/common/oauth/KeystoreHelper.java

```diff
@@ -73,17 +73,21 @@ public SignatureHelper getSignatureHelper() {
         }
     }
 
-    private KeyStore getKeystoreEntry(final InputStream i) throws IOException, RuntimeException, NoSuchAlgorithmException, CertificateException, KeyStoreException {
+    private KeyStore getKeystoreEntry(final InputStream i, final boolean certificate) throws IOException, RuntimeException, NoSuchAlgorithmException, CertificateException, KeyStoreException {
         KeyStore keystore = KeyStore.getInstance("JKS");
         keystore.load(i, this.keystore.getStorePassword().toCharArray());
         if (!keystore.containsAlias(this.keystore.getAlias())) {
-            throw new RuntimeException("no key with alias " + this.keystore.getAlias() + " found in the keystore "
+            throw new RuntimeException("no entry with alias " + this.keystore.getAlias() + " found in the keystore "
                     + this.keystore.getLocation());
         }
-        if (keystore.isKeyEntry(this.keystore.getAlias())) {
+        if (!certificate && keystore.isKeyEntry(this.keystore.getAlias())) {
             throw new RuntimeException("no key with alias " + this.keystore.getAlias() + " found in the keystore "
                     + this.keystore.getLocation());
         }
+        if (certificate && keystore.isCertificateEntry(this.keystore.getAlias())) {
+            throw new RuntimeException("no certificate with alias " + this.keystore.getAlias() + " found in the keystore "
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS00112)